src/renderer/gl_engine/tvgGlRenderer.cpp

```diff
@@ -1048,10 +1048,7 @@ bool GlRenderer::renderShape(RenderData data)
     {
         auto gradient = sdata->rshape->fill;
         if (gradient) drawPrimitive(*sdata, gradient, RenderUpdateFlag::Gradient, drawDepth1);
-    }
-
-    if(flags & (RenderUpdateFlag::Color | RenderUpdateFlag::Transform))
-    {
+    } else if(flags & (RenderUpdateFlag::Color | RenderUpdateFlag::Transform)) {
```

Comment:
wasn't this?
```
if (transform) {
   if (gradient fill)
   else (solid fill)
   
   if (gradient fill stroke)
   else (solid fill stroke)
}
```

Reply:
Can I only check RenderUpdateFlag::Color and RenderUpdateFlag::Gradient here?

Reply:
> Can I only check RenderUpdateFlag::Color and RenderUpdateFlag::Gradient here?

yes, come to see again, my comment is wrong. :)

my intention was like: 
```
if (gradient fill | solid fill | transform) 
   if (gradient fill)...
   else (solid fill) ...
```